src/Tools/ExternalAccess/OmniSharp/Workspace/OmniSharpDocumentId.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Workspace
+{
+    internal static class OmniSharpDocumentId
+    {
+        public static DocumentId CreateFromSerialized(ProjectId projectId, Guid id, bool isSourceGenerated, string? debugName)
```

Comment:
what's the use case where omnisharp needs to *create* a doc-id from a serialized form?  How does it even get the value to pass into `bool isSourceGenerated`?  we don't expose that property public on our doc-id in the first place.   Thanks! :)

Reply:
The O# navigation service can return source generated file info as part of a GoToDefinition response (See [omnisharp/omnisharp-roslyn/GoToDefinitionServiceV2.cs](https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn.CSharp/Services/Navigation/GotoDefinitionServiceV2.cs#L68)). When this navigation item is selected by the user the client will then request the document from the server and provide it to the editor (See [dotnet/vscode-csharp/SourceGeneratedDocumentProvider.ts](https://github.com/dotnet/vscode-csharp/blob/main/src/features/sourceGeneratedDocumentProvider.ts)).

Reply:
i don't understand.  you are undoing things to get the guid and pass that along.  instead of doing that, why not actually send the DocID itself instead?  i.e. instead of `document.Id.Id`, just send `doument.Id`.

Reply:
@CyrusNajmabadi I think the case here is similar to what we have for LSP: we can take the document ID as a GUID and send that over the wire, and when we get it back we know it's source generated based on the URI prefix or something similar.